workspace-grid@mathematical.coffee.gmail.com/extension.js

```diff
@@ -1116,7 +1157,12 @@ function unoverrideWorkspaceDisplay() {
 * Sets org.gnome.shell.overrides.dynamic-workspaces schema to false
 *******************/
 function disableDynamicWorkspaces() {
-    let settings = global.get_overrides_settings();
+    let settings;
+    // Override schemas are gone in GNOME 3.30
+    if (global.get_overrides_settings)
```

Comment:
Same here. Can't we check the current version and decide how to proceed?

Reply:
That should be better now:

```diff
diff --git a/workspace-grid@mathematical.coffee.gmail.com/extension.js b/workspace-grid@mathematical.coffee.gmail.com/extension.js
index 7f3e2048..077f937 100644
--- a/workspace-grid@mathematical.coffee.gmail.com/extension.js
+++ b/workspace-grid@mathematical.coffee.gmail.com/extension.js
@@ -355,7 +355,7 @@ function showWorkspaceSwitcher(display, arg2, arg3, arg4) {
     let window;
     let binding;
     // Note: in v3.30, the 2nd arg (screen that we don't need) has been removed
-    if (!arg4) {
+    if (Utils.isVersionAbove(3, 28)) {
         screen = display.get_workspace_manager();
         window = arg2;
         binding = arg3;
@@ -1159,10 +1159,11 @@ function unoverrideWorkspaceDisplay() {
 function disableDynamicWorkspaces() {
     let settings;
     // Override schemas are gone in GNOME 3.30
-    if (global.get_overrides_settings)
-        settings = global.get_overrides_settings()
-    else
+    if (Utils.isVersionAbove(3, 28)) {
         settings = new Gio.Settings({ schema_id: 'org.gnome.mutter' });
+    } else {
+        settings = global.get_overrides_settings()
+    }
     settings.set_boolean('dynamic-workspaces', false);
 }
 
```